docs/docs/guides/labs/dg/incrementally-adopting-dg/migrating-project.md

```diff
@@ -7,61 +7,85 @@ import Preview from '@site/docs/partials/\_Preview.md';
 
 <Preview />
 
-:::note
-
-This guide is only relevant if you are starting from an _existing_ Dagster project. This setup is unnecessary if you [scaffolded a new project](/guides/labs/dg/scaffolding-a-project) with `dg scaffold project`.
-
-:::
+Suppose we have a basic existing Dagster project. Our project defines a Python
+package with a single Dagster asset. The asset is exposed in a top-level
+`Definitions` object in `my_existing_project/definitions.py`. We'll consider
+both a case where we have been using [uv](https://docs.astral.sh/uv/) with `pyproject.toml` and [`pip`](https://pip.pypa.io/en/stable/) with `setup.py`.
+
+<Tabs groupId="package-manager">
+    <TabItem value="uv" label="uv">
+        <CliInvocationExample path="docs_snippets/docs_snippets/guides/dg/migrating-project/1-uv-tree.txt" />
+    </TabItem>
+    <TabItem value="pip" label="pip">
+        <CliInvocationExample path="docs_snippets/docs_snippets/guides/dg/migrating-project/1-pip-tree.txt" />
+    </TabItem>
+</Tabs>
 
-We have a basic existing Dagster project and have been using `pip` to manage
-our Python environment. Our project defines a Python package with a `setup.py`
-and a single Dagster asset. The asset is exposed in a top-level `Definitions`
-object in `my_existing_project/definitions.py`.
+`dg` works best with using project-scoped virtual environments.
+Before proceeding, let's make sure you have an activated and up-to-date virtual
+environment in the project root.
+
+<Tabs groupId="package-manager">
+    <TabItem value="uv" label="uv">
+        If you don't have a virtual environment yet, run:
+        <CliInvocationExample contents="uv sync" />
+        Then activate it:
+        <CliInvocationExample contents="source .venv/bin/activate" />
+    </TabItem>
+    <TabItem value="pip" label="pip">
+        If you don't have a virtual environment yet, run:
+        <CliInvocationExample contents="python -m venv .venv" />
+        Now activate it:
+        <CliInvocationExample contents="source .venv/bin/activate" />
+        And install the project package as an editable install:
+        <CliInvocationExample contents="pip install --editable ." />
+    </TabItem>
+</Tabs>
```

Comment:
Should we make it more clear here that the only actual requirement is being able to import the project as a module? I feel like this is closer to sounding like the subdir venv is required to migrate. 

I think the migration guide is one of the more likely places someone will be coming with an existing venv setup that they have been using for their project so far

Reply:
Changed to:

```
`dg` needs to be able to resolve a Python environment for your project. This
environment must include an installation of your project package. By default,
a project's environment will resolve to whatever virtual environment is
currently activated in the shell, or system Python if no virtual environment is
activated.

Before proceeding, we'll make sure we have an activated and up-to-date virtual
environment in the project root. Having the virtual environment located in the
project root is recommended (particularly when using `uv`) but not required.

```